modules/3d/include/opencv2/3d.hpp

```diff
@@ -2342,6 +2341,167 @@ class CV_EXPORTS CvLevMarq
+Each node contains 8 children, which are used to divide the space cube into eight parts. Each octree node represents
+a cube. And these eight children will have a fixed order, the order is described as follows:
+
+    For illustration, assume,
+        rootNode: origin == (0, 0, 0), size == 2
+     Then,
+        children[0]: origin == (0, 0, 0), size == 1
+        children[1]: origin == (1, 0, 0), size == 1, along X-axis next to child 0
+        children[2]: origin == (0, 1, 0), size == 1, along Y-axis next to child 0
+        children[3]: origin == (1, 1, 0), size == 1, in X-Y plane
+        children[4]: origin == (0, 0, 1), size == 1, along Z-axis next to child 0
+        children[5]: origin == (1, 0, 1), size == 1, in X-Z plane
+        children[6]: origin == (0, 1, 1), size == 1, in Y-Z plane
+        children[7]: origin == (1, 1, 1), size == 1, furthest from child 0
+*/
+class Octree {
+
+public:
+
+    //! Default constructor.
+    Octree();
+
+    /** @overload
+        * @brief Create an empty Octree and set the maximum depth.
+        * @param _maxDepth The max depth of the Octree. The _maxDepth > -1.
+        */
+    Octree(int _maxDepth);
+
+    /** @overload
+        *  @brief Create an Octree from the PointCloud data with the specific max depth.
+        * @param _maxDepth The max depth of the Octree.
+        * @param _pointCloud Point cloud data.
+        */
+    Octree(int _maxDepth, std::vector<Point3f> &_pointCloud);
+
+    /** @overload
+        * @brief Create an empty Octree.
+        * @param _maxDepth Max depth.
+        * @param _size Initial Cube size.
+        * @param _origin Initial center coordinate.
+        */
+    Octree(int _maxDepth, double _size, Point3f _origin);
+
+    //! Default destructor
+    ~Octree();
+
+    /** @brief Insert a point data to a OctreeNode.
+        *
+        * @param point The point data in Point3f format.
+        */
+    void insertPoint(Point3f point);
+
+    /** @brief Read point cloud data and create OctreeNode.
+        * This function is only called when the octree is being created.
+        * @param pointCloud PointCloud data.
+        * @return Returns whether the creation is successful.
+        */
+    bool convertFromPointCloud(std::vector<Point3f> &pointCloud);
+
+    /** @brief Determine whether the point is within the space range of the specific cube.
+        *
+        * @param point The point coordinates.
+        * @return If point is in bound, return ture. Otherwise, false.
+        */
+    bool isPointInBound(Point3f point);
+
+    //! Set MaxDepth for Octree.
+    void setMaxDepth(int maxDepth);
+
+    //! Set Box Size for Octree.
+    void setSize(double _size);
+
+    //! Set Origin coordinates for Octree.
+    void setOrigin(Point3f _origin);
+
+    //! returns true if the rootnode is NULL.
+    bool empty();
+
+    /** @brief
+        *  Reset all octree parameterDeleting a point from the octree actually deletes the corresponding element
```

Comment:
> parameterDeleting

what is that?

Please re-read documentation.

Reply:
fixed